tests/test_compare_db.py

```diff
@@ -322,9 +321,55 @@ def test_batch_sqlite_exception_insert_only(db):
     """Should rollback even if we don't start the explicit transaction in match()"""
     batch = db.batch()
     batch.insert_orig(100, name="Test")
-    batch.insert_orig(("bogus",), name="Test")
+    batch.insert_orig(2**100, name="Test")
 
-    with pytest.raises(sqlite3.Error):
+    with pytest.raises(OverflowError):
         batch.commit()
 
     assert db.get_by_orig(100) is None
+
+
+def test_batch_insert_with_invalid_address(db):
+    """Should not create (insert functions) an entity if the address parameter is invalid."""
+    with db.batch() as batch:
+        batch.insert_orig(None, name="Test")
```

Comment:
Now that I think about it again, I am genuinely a bit confused why `mypy` isn't complaining here

Reply:
Ah, here we go: It's the untyped `db` parameter, which is treated as an `Any` because our mypy config isn't exactly strict:
![grafik](https://github.com/user-attachments/assets/1f7c2a01-1492-4c30-9071-f55c0cd48d61)